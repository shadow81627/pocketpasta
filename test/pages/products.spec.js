import Component from '@/pages/products/index.vue';
import productData from '@/assets/link-data/products/tomatoes.json';
import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
import BootstrapVuePlugin from 'bootstrap-vue';

const localVue = createLocalVue();

const product = delete productData['@context'];

localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(Vuetify);
localVue.use(BootstrapVuePlugin);

const router = new VueRouter();

let getters;
let state;
let store;
let mutations;

beforeEach(() => {
  state = {
    products: [{ ...product, id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  };

  getters = {
    getProductById: (state) => (id) =>
      state.products.find((product) => product.id === id),
  };

  mutations = {
    setTheme(currentState, theme) {
      currentState.theme = theme;
    },
  };

  store = new Vuex.Store({
    state,
    getters,
    mutations,
  });
});

const factory = () =>
  shallowMount(Component, {
    store,
    router,
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    localVue,
  });

describe('products page', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(Component.head()).toBeTruthy();
  });
});
