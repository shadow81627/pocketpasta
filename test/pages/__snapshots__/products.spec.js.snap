// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products page renders properly 1`] = `
"<div class=\\"container\\">
  <list-stub collection=\\"products\\" heading=\\"products.heading\\" layout=\\"list\\" fetchonserver=\\"true\\" defaultlimit=\\"-1\\" infinite=\\"true\\" headers=\\"[object Object],[object Object],[object Object]\\"></list-stub>
</div>"
`;
